tests/FakeItEasy.Tests/Creation/CastleDynamicProxy/CastleDynamicProxyGeneratorTests.cs

```diff
@@ -207,7 +212,11 @@ public void Should_specify_that_no_default_constructor_was_found()
             result.ReasonForFailure.Should().StartWith("No usable default constructor was found on the type");
         }
 
+#if !NETCORE
         [Fact]
+#else
+        [Fact(Skip = "System.AppDomainInitializerInfo is not available in .NET Core. Furthermore, even if we could found another non-static internal class in mscorlib that is type-forwarded to Private.CoreLib on .NET Core it may disappear in the future too.")]

```

Comment:
The only reason I'd say we'd need an external assembly is that many of our tests have `InternalsVisibleTo` DynamicProxyGenAssembly2. We need to talk about a type that's
1. in an assembly referenced by the the assembly this test belongs to, 
2. is internal, and
3. is not exposed to DynamicProxyGenAssembly2

we currently reference 

![image](https://cloud.githubusercontent.com/assets/3275797/16629113/96b7fd9c-4382-11e6-87ff-48fe167e5531.png)

So we could search out an acceptable type not in FakeItEasy (with the risk that it would go away) or reference another assembly, one which we have control over. In that case we could make one.
This is perhaps not ideal, but in the short term, I'd be inclined to look for an internal type in something we already reference. We wouldn't be making the situation worse!